build.gradle

```diff
@@ -4,108 +4,143 @@ buildscript {
+targetCompatibility=11
+assert System.properties["java.specification.version"] == "11" || "12" || "13"
+
+apply from: "gradle/docker.gradle"
+apply from: "gradle/sonar.gradle"
+//jhipster-needle-gradle-apply-from - JHipster will add additional gradle scripts to be applied here
+
+if (project.hasProperty("prod")) {
+    apply from: "gradle/profile_prod.gradle"
+} else {
+    apply from: "gradle/profile_dev.gradle"
 }
-defaultTasks 'bootRun'
 
-bootWar {
-   mainClassName = 'com.icthh.xm.gate.GateApp'
+if (project.hasProperty("war")) {
+    apply from: "gradle/war.gradle"
 }
 
-war {
+if (project.hasProperty("zipkin")) {
+    apply from: "gradle/zipkin.gradle"
+}
 
+idea {
+    module {
+        excludeDirs += files("node_modules")
+    }
 }
 
-springBoot {
-    mainClassName = 'com.icthh.xm.gate.GateApp'
-    buildInfo()
+eclipse {
+    sourceSets {
+        main {
+            java {
+                srcDirs += ["build/generated/sources/annotationProcessor/java/main"]
+            }
+        }
+    }
 }
 
+// See https://virgo47.wordpress.com/2018/09/14/classpath-too-long-with-spring-boot-and-gradle/ for details
+// https://github.com/jhipster/generator-jhipster/issues/9713
 if (OperatingSystem.current().isWindows()) {
-    // https://stackoverflow.com/questions/40037487/the-filename-or-extension-is-too-long-error-using-gradle
     task classpathJar(type: Jar) {
-        dependsOn configurations.runtime
-        appendix = 'classpath'
+        inputs.files sourceSets.main.runtimeClasspath
 
+        archiveName = "runboot-classpath.jar"
         doFirst {
             manifest {
-                attributes 'Class-Path': configurations.runtime.files.collect {
-                    it.toURI().toURL().toString().replaceFirst(/file:\/+/, '/').replaceAll(' ', '%20')
-                }.join(' ')
+                def classpath = sourceSets.main.runtimeClasspath.files
+                attributes "Class-Path": classpath.collect {f -> f.toURI().toString()}.join(" ")
             }
         }
     }
 
     bootRun {
-        dependsOn classpathJar
-        doFirst {
-            classpath = files("$buildDir/classes/java/main", "$buildDir/resources/main", classpathJar.archivePath)
-        }
+        classpath = classpathJar.outputs.files
     }
 }
 
+defaultTasks "bootRun"
+
+bootWar {
+    mainClassName = 'com.icthh.xm.gate.GateApp'
+}
+
+war {
+
+}
+
+springBoot {
+    mainClassName = "com.icthh.xm.gate.GateApp"
+}
+
 test {
-    systemProperty "spring.profiles.active", "test"
-    include '**/*UnitTest*'
-    include '**/*IntTest*'
+    useJUnitPlatform()
+    exclude "**/*IT*", "**/*IntTest*"
 
+    testLogging {
+        events 'FAILED', 'SKIPPED'
+    }
     // uncomment if the tests reports are not generated
     // see https://github.com/jhipster/generator-jhipster/pull/2771 and https://github.com/jhipster/generator-jhipster/pull/4484
     // ignoreFailures true
     reports.html.enabled = false
+}
+
+task integrationTest(type: Test) {
```

Comment:
remove this gradle task, and copy "test" section from old project, because this task doesn't execute our tests